bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/dialogs/SearchPattern.java

```diff
@@ -549,6 +630,22 @@ public final int getMatchRule() {
 		return this.matchRule;
 	}
 
+	/**
```

Comment:
Can't we do without those 2 new APIs? as we have access to the rule one can query it with bit masks. We try to minimize the amount of API we create/maintain in Platform.

Reply:
Well, I see your point / goal, but I'm not quite sure here. If we were to "query the pattern's rule with bit mask", I think the code is not ready for this scenario yet? I.e. a) there is no `RULE_SUFFIX_MATCH` to use for such a query, b) just one of the `RULE_*` constants is set into the inferred `matchRule` field now, so it doesn't expect to be a combination of multiple flags.

Reply:
We don't really care about a `RULE_SUFFIX_MATCH`; also I wonder can't pattern already deal with suffix? Why do you need particular consideration of suffix here; can't this be translated into a regular pattern without extra state?
Anyway the javadoc of `getMatchRule` says we can have the rule combined using bit masks.
In any case, it would be great if we can just keep this methods private, or package protected.

Reply:
I have updated the javadocs in a the new commit to better describe the reality. I think they were historically wrong, so you might have also been mislead by them...

> Can't we do without those 2 new APIs?

I will recheck, it will probably require to clone the suffix-checking code into the client code (which is currently just the Resource dialog)...

Reply:
@mickaelistria, so I have "moved" the new `isMatch...()` methods into the Dialog code in the new commit. Can you please check it out? It can't be probably done much simpler, provided we want to keep all the changes small?

Reply:
Seems like accepted, closing thread.